dspace/bin/dspace

```diff
@@ -12,20 +12,18 @@
 
 BINDIR=`dirname $0`
 DSPACEDIR=`cd "$BINDIR/.." ; pwd`
-# Add the directory with all oai classes (original and overlay) needed by the 'dspace oai' launcher.xml
-OAICLASSES=$DSPACEDIR/webapps/oai/WEB-INF/classes/
 
 # Get the JARs in $DSPACEDIR/jsp/WEB-INF/lib, separated by ':'
-JARS=`echo $DSPACEDIR/lib/*.jar | sed 's/ /\:/g'`
+JARS="$DSPACEDIR/lib/*"
```

Comment:
This gives a space-separated list, not the colon-separated path-list needed by -classpath.
The ``sed`` command could be replaced with ``tr ' ' ':'``

Reply:
That is correct, but what about spaces in folder names? They would be replaced as well, causing Jars to not appear on the classpath.

Reply:
True:  we can't use simple string replacement.  But we must have a colon-separated list.
The path will have to be built iteratively, like this:
```
unset JARS
for JAR in ${DSPACEDIR}/lib/*
do
    [ -v JARS ] && JARS=${JARS}:
    JARS=${JARS}${JAR}
done
```
and then put doublequotes around $FULLPATH to protect any interior spaces.

Reply:
I did some testing and found (as I should have known) that `JARS="$DSPACEDIR/lib/*" is not expanded, so we would be passing the glob expression into `java` as the classpath.  At least in OpenJDK, the interpreter seems to be expanding the glob itself, contrary to its own documentation.  I recommend that we not depend on undocumented behavior.